workflow/scripts/rule_sample_mask.py

```
@@ -0,0 +1,24 @@
+
+ds_dir = snakemake.input[0].replace(".vcf_done", "")
+ds = sgkit.load_dataset(ds_dir)
+with open(snakemake.input[-1]) as f:
+    sample_ids = numpy.genfromtxt(f, dtype=str)
+sample_ids = xarray.DataArray(sample_ids, dims="sample")
+sample_mask = ds.sample_id.isin(sample_ids)
+if sample_mask.sum() != len(sample_ids):
+    raise ValueError(
+        f"Could not find all samples in dataset. "
+        f"Failed to find {sample_ids[~sample_mask].values}"
+    )
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)